onnxscript/version_converter/_version_converter.py

```diff
@@ -276,39 +277,45 @@ def visit_node(
             self.replace_node(node, replacement, root)
         return None
 
-    def visit_graph(self, graph: ir.Graph) -> None:
+    def visit_graph(self, graph: ir.Graph) -> ir.Graph | None:
         if self.target_version > SUPPORTED_MAX_ONNX_OPSET:
             logger.warning(
                 "Conversion to target opset: %s not currently supported.",
                 self.target_version,
             )
             return None
-        for node in graph:
-            up_conversion = True
-            if node.version is None:
-                node.version = self.model_version
-            # Iterate each node from current node version -> target version
-            # and updating node based on the correct adapter
-            # Up-conversion [ver->ver+1] or down-conversion [ver->ver-1]
-            # TODO(shubhambhokare1): Remove once down-conversion adapters are supoorted
-            if self.target_version < node.version:
-                up_conversion = False
-                logger.warning(
-                    "Target opset: %s less than %s, downstream version conversion not currently handled.",
-                    self.target_version,
-                    self.model_version,
-                )
-                return None
-            for opset_version in range(node.version, self.target_version):
+
+        # TODO(shubhambhokare1): Support down-conversion
+        while self.model_version < self.target_version:
+            pre_conversion_graph = copy.copy(graph)
+            # Up-convert each node in the graph from opset_version -> opset_version + 1
+            # or down-convert from opset_version -> opset_version - 1
+            # Return non-converted graph if any node fails to convert.
+            for node in graph:
+                up_conversion = True
+                if node.version is None:
```

Comment:
We should check if it is an onnx domain op first: else skip, and don't do anything (even updating node.version).